core-bundle/src/EventListener/DataContainer/RootPageDependentModuleListener.php

```diff
@@ -0,0 +1,206 @@
+    private Connection $connection;
+    private TranslatorInterface $translator;
+    private ScopeMatcher $scopeMatcher;
+    private RequestStack $requestStack;
+    private CsrfTokenManagerInterface $csrfTokenManager;
+    private string $csrfTokenName;
+
+    public function __construct(Connection $connection, TranslatorInterface $translator, ScopeMatcher $scopeMatcher, RequestStack $requestStack, CsrfTokenManagerInterface $csrfTokenManager, string $csrfTokenName)
+    {
+        $this->connection = $connection;
+        $this->translator = $translator;
+        $this->scopeMatcher = $scopeMatcher;
+        $this->requestStack = $requestStack;
+        $this->csrfTokenManager = $csrfTokenManager;
+        $this->csrfTokenName = $csrfTokenName;
+    }
+
+    /**
+     * @Hook("loadDataContainer")
+     */
+    public function onLoadDataContainer(string $table): void
+    {
+        /** @var Request $request */
+        $request = $this->requestStack->getCurrentRequest();
+
+        if (!$request instanceof Request || !$this->scopeMatcher->isBackendRequest($request)) {
+            return;
+        }
+
+        $fields = $GLOBALS['TL_DCA'][$table]['fields'];
+
+        if (
+            !\is_array($fields) ||
+            !\in_array('rootPageDependentModule', array_column($fields, 'inputType'), true)
+        ) {
+            return;
+        }
+
+        foreach ($fields as $key => $field) {
+            if (!\array_key_exists('inputType', $field)) {
+                unset($fields[$key]);
+            }
+        }
+
+        $affectedFields = array_keys(
+            array_column($fields, 'inputType'),
+            'rootPageDependentModule',
+            true
+        );
+
+        foreach ($affectedFields as $affectedField) {
+            $field = \array_slice($fields, $affectedField, 1);
+            $keys = array_keys($field);
+
+            $key = $keys[0];
+            $fieldConfig = $GLOBALS['TL_DCA'][$table]['fields'][$key];
+
+            if (!\array_key_exists('eval', $fieldConfig)) {
+                $fieldConfig['eval'] = [];
+            }
+
+            if (!\array_key_exists('rootPages', $fieldConfig['eval'])) {
+                $fieldConfig['eval']['rootPages'] = $this->getRootPages();
+            }
+
+            if (!\array_key_exists('blankOptionLabel', $fieldConfig['eval'])) {
+                $fieldConfig['eval']['blankOptionLabel'] = $this->translator->trans(
+                    sprintf('tl_module.%sBlankOptionLabel', $key),
+                    [],
+                    'contao_module'
+                );
+            }
+
+            // Save modified configuration back to global DCA
+            $GLOBALS['TL_DCA'][$table]['fields'][$key] = $fieldConfig;
+        }
+    }
+
+    public function onOptionsCallback(DataContainer $dc): array
+    {
+        $options = [];
+        $types = $GLOBALS['TL_DCA'][$dc->table]['fields'][$dc->field]['eval']['modules'] ?? [];
+
+        $statement = $this->connection->executeQuery(
```

Comment:
`$statement` will likely get overwritten a few lines down, might be better suited in an else clause. :upside_down_face:

Reply:
It's actually ment to be overwritten if there is some types available to filter for. Possibly more a personal thing, as I don't like else clauses 🙈

Reply:
Refactored in 5c9f0a50b.